### DISCUSSION TITLE
How can I access the request headers on an error page?

### DISCUSSION THREAD

**Discussion by thijsw** [nuxt/nuxt#16788] [Questions]
I created an `error.vue` page to show errors. On this page, I'd like to have access to the request headers, so that I can customize the error page based on the host and or cookie values. When I try to access these values using the useRequestHeaders() composable, it returns an empty object. I'm not sure whether this is a bug or intended behavior; could anybody explain how I could make this work?

app.vue
```vue
<template>
  <div>This page throws an error</div>
</template>

<script lang="ts" setup>
throw new Error();
</script>
```

error.vue
```vue
<template>
  <div>{{ headers }}</div>
</template>

<script lang="ts" setup>
const headers = useRequestHeaders();
</script>
```

result:
```html
{}
```
https://stackblitz.com/edit/nuxt-starter-j5z4sw

**Comment by sevillaarvin** [comment]
Based on this https://github.com/nuxt/framework/discussions/7761 you can use `useState`.
